src/solver/simulation/hydro-final-reservoir-level-functions.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+** Copyright 2007-2018 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#include "hydro-final-reservoir-level-functions.h"
+#include <antares/emergency.h>
+
+namespace Antares::Solver
+{
+static void initializeGeneralData(Data::FinalReservoirLevelRuntimeData& finLevData,
+                           const Data::Parameters& parameters,
+                           uint year)
+{
+    finLevData.simEndDay = parameters.simulationDays.end;
+    finLevData.yearIndex = year;
+}
+
+static void initializePerAreaData(Data::FinalReservoirLevelRuntimeData& finLevData,
```

Comment:
Make this a member function of FinalReservoirLevelRuntimeData ?

Reply:
Ah, remember this now. Tried to do it like that originally.
However, cannot pass "area" as a function parameter if the method is defined in FinalReservoirLevelRuntimeData.
Always get error: invalid use of incomplete type ‘const class Antares::Data::Area’
Tried with the forward declaration in the container.h but did not help.